src/schemas/users.ts

```diff
@@ -0,0 +1,6 @@
+import { Type } from '@sinclair/typebox'
+
+export const UpdateCredentialsSchema = Type.Object({
+  currentPassword: Type.String(),
+  newPassword: Type.String()
```

Comment:
I would force to use a strong password and check that the two values are different.